### ISSUE TITLE
problems about CNN architecture initialization

### ISSUE THREAD

**Issue by fengyunxiaozi** [vlfeat/matconvnet#300]
hi, in the examples provided by MatConvNet, the code 'cnn_mnist_init'  is used to initialize the mnist cnn, i was confused about this code. I mean why the initialized architecture in this code is "conv-pool-conv-pool-conv-relu..." rather than  "conv-relu-pool-conv-relu-pool-conv-relu...".  Because relu is not a linear operator, the two initialized ways should be different.  in addition, I was also confused about another problem, if this style "conv-relu-pool..." is equivalent to "conv-pool-relu...".. thanks in advance !!

**Comment by fengyunxiaozi** [comment]
@vedaldi @lenck

**Comment by vedaldi** [comment]
Hi, there is no unique way of defining architectures. Some use pool before relu, some do the opposite, some do not have pool, etc.

They are most definitely not functionally equivalent, but some combinations tend to result in similar performance.

As for the MNIST example, we simply used the same definition for the MNIST example in Caffe. I *believe* that this is similar to how the original LeNet architecture was defined many years ago, but I did not really check carefully. Again, just one of the many architecture variants.
